### DISCUSSION TITLE
How to add the tracking id over the detected object?

### DISCUSSION THREAD

**Discussion by raj-kapil** [ultralytics/ultralytics#3268] [Q&A]
I am using the default track method with supervision library 
```
results = model.track(source=INPUT_VIDEO_PATH, conf=0.8, save=True,  tracker="bytetrack.yaml", persist=True)
```

How can I add the tracking id over the bounding box in the output video?

This code should have added the tracking id by default over the bounding boxes
```
if pred_boxes and show_boxes:
    for d in reversed(pred_boxes):
        c, conf, id = int(d.cls), float(d.conf) if conf else None, None if d.id is None else int(d.id.item())
        name = ('' if id is None else f'id:{id} ') + names[c]
        label = (f'{name} {conf:.2f}' if conf else name) if labels else None
        annotator.box_label(d.xyxy.squeeze(), label, color=colors(c, True))
``` 

but it is not adding that, when I am extracting the boxes ids and doing the custom labelling then it works. Here's my code snippet for the same,
```
with VideoSink(TARGET_VIDEO_PATH, video_info) as sink:
    results = model.track(source=INPUT_VIDEO_PATH, conf=0.8, save=True,  tracker="bytetrack.yaml", persist=True)
    for result in results:
        frame = result.orig_img
        detections = sv.Detections.from_yolov8(result)
        if result.boxes.id is not None:
                detections.tracker_id = result.boxes.id.cpu().numpy().astype(int)
                labels = [
                    f"#{tracker_id} {model.model.names[class_id]} {int(100*confidence)}%"
                    for box_coord, _, confidence, class_id, tracker_id
                    in detections
                ]
                frame = box_annotator.annotate(
                    scene=frame, 
                    detections=detections,
                    labels=labels
                )
                line_counter.trigger(detections=detections)
                line_annotator.annotate(frame=frame, line_counter=line_counter)

```

**Comment by glenn-jocher** [comment]
@raj-kapil it seems like you're on the right track with using the `track` method for object tracking. The tracking ID should indeed be added to the bounding boxes by default when using the `track` method. If the tracking IDs are not appearing as expected, it could be due to a configuration issue or a bug.

Since you've mentioned that custom labeling with your own code snippet works, it suggests that the tracking IDs are being generated but not displayed. This could be due to the way the `annotator.box_label` function is being called or a problem with the `pred_boxes` data structure.

I would recommend reviewing the configuration of your tracker and ensuring that the `show_boxes` flag is set correctly. Also, make sure that the `annotator.box_label` function is being called after the tracking IDs are assigned.

If the issue persists, please provide a minimal reproducible example on the GitHub issue tracker, and we'll look into it further. Remember to include details about your environment and any relevant configurations. Our team is committed to improving the YOLOv8 experience, and your feedback is valuable in helping us identify and fix such issues. 🛠️

For more detailed guidance on tracking and other features, please refer to our documentation at https://docs.ultralytics.com.
